content/fr/articles/js/es2015/classes.md

```diff
@@ -0,0 +1,148 @@
+---
+date: "2015-12-08"
+title: "ES6, ES2015 : les classes"
+tags:
+  - javascript
+  - ES6
+  - ES2015
+authors:
+  - magsout
+---
+
+Les classes introduites par `ES2015` ne sont clairement que du sucre syntaxique.
+Elles n'apportent rien de plus en terme de fonctionnalité. Toutefois, l'objectif
+de cette évolution est de rendre le code plus lisible et plus facilement
+accessible.
+
+## Déclaration de classes
+
+Oubliez les fonctions anonymes et les prototypes, la sémantique a pris le
+relais, comme vous pouvez le constater dans l'exemple de déclaration de la
+classe `User` :
+
+```js
+class User {
+  // méthode constructor
+  constructor(firstname, lastname) {
+    this.firstname = firstname;
+    this.lastname = lastname;
+  }
+
+  sayName() {
+    return `${this.firstname} ${this.lastname}`
+  }
+}
+
+// instanciation
+const user = new User("John", "Doe")
+
+// appel de la méthode sayName()
+console.log(user.sayName()) // John Doe
+```
+
+Pour rappelle, voici une [façon possible
+d'écrire](https://gist.github.com/magsout/a876b2fa8240a987e523) cette classe en
+`es5` :
+
+```js
+var User = function(firstname, lastname) {
```

Comment:
as `arguments.callee` isn't present in strict mode, that could be a bit confusing.